code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -425,18 +411,31 @@
 
 		if("transfer")
 			if(is_printing)
-				say("buffer locked while printing!")
+				say("The buffer is locked while printing.")
 				return
 
 			var/reagent_ref = params["reagentRef"]
 			var/amount = params["amount"]
 			var/target = params["target"]
 
+			if(amount == -1) // Set custom amount
+				amount = tgui_input_number(ui.user, "Enter amount to transfer", "Transfer amount")
```

Comment:
```suggestion
				var/mob/user = ui.user //Hold a reference to the user if the UI is closed
				amount = tgui_input_number(user, "Enter amount to transfer", "Transfer amount")
				if(!amount || !user.can_perform_action(src))
					return FALSE
```

Reply:
> `if(!amount)`

it's checked later in transfer_reagents proc but I suppose theres' no harm in duplicating that check